src/ports/web/api/events/export.rs

```diff
@@ -0,0 +1,176 @@
+
+    use icalendar::{Calendar, Class, Component, Event, Property};
+
+    let mut calendar = events
+        .map(|ev| {
+            let mut e = Event::new()
+                .uid(ev.id.as_str())
+                .summary(&ev.title)
+                .starts(ev.start)
+                .class(Class::Public)
+                // https://icalendar.org/iCalendar-RFC-5545/3-8-1-2-categories.html
+                .append_property(Property::new("CATEGORIES", &ev.tags.join(",")).done())
+                .done();
+
+            if let Some(l) = &ev.location {
+                // https://icalendar.org/iCalendar-RFC-5545/3-8-1-6-geographic-position.html
+                e = e
+                    .append_property(
+                        Property::new(
+                            "GEO",
+                            &format!("{};{}", l.pos.lat().to_deg(), l.pos.lng().to_deg()),
+                        )
+                        .done(),
+                    )
+                    .done();
+
+                if let Some(addr) = address_from_location(l) {
+                    // https://icalendar.org/iCalendar-RFC-5545/3-8-1-6-geographic-position.html
+                    e = e
+                        .append_property(Property::new("LOCATION", &addr).done())
+                        .done();
+                }
+            }
+
+            if let Some(contact) = contact_from_ev(&ev) {
+                // https://icalendar.org/iCalendar-RFC-5545/3-8-4-2-contact.html
+                e = e
+                    .append_property(Property::new("CONTACT", &contact).done())
+                    .done();
+            }
+            if let Some(end) = ev.end {
+                e = e.ends(end).done();
+            }
+            if let Some(desc) = ev.description {
+                e = e.description(&desc).done();
+            }
+            e
+        })
+        .collect::<Calendar>();
+    let calendar = calendar.name("OpenFairDB events").done();
+    let data = calendar.to_string();
+    let c_type = ContentType::new("text", "calendar");
+    Ok(Content(c_type, data))
+}
+
+fn address_from_location(l: &Location) -> Option<String> {
+    l.address.as_ref().and_then(|a| {
+        if a.is_empty() {
+            None
+        } else {
+            let mut details = vec![];
+            if let Some(street) = &a.street {
+                details.push(street.clone());
+            }
+            if let Some(zip) = &a.zip {
+                details.push(zip.clone());
+            }
+            if let Some(city) = &a.city {
+                details.push(city.clone());
+            }
+            if let Some(country) = &a.country {
+                details.push(country.clone());
+            }
+            if let Some(state) = &a.state {
+                details.push(state.clone());
+            }
+            Some(details.join("\\, "))
+        }
+    })
+}
+
+fn contact_from_ev(ev: &Event) -> Option<String> {
```

Comment:
In function names abbreviations should be used even more restrictive! -> contact_from_event()